resolver/resolver_test.go

```diff
@@ -85,6 +88,124 @@ func TestResolverTTL(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+func TestAddressFamilyAffinity(t *testing.T) {
```

Comment:
The fake `net.Resolver` part is actually pretty straightforward imo, and at 72 lines including error handling and other boilerplate it's not really that long. Would be nicer if you could flat-out fake results, but the two ways to do that are internal to Go standard library only.

You're right, though, this could just be done by making the affinity handling its own function. Probably should do that.

Do we really want to delete these tests entirely, though? I actually quite like having a proper end-to-end test of the DNS resolver that is somewhat less trivial. Since it really does everything as if we're in the real world, it shouldn't be very fragile either.

Reply:
Sure, leave 'em. I guess it was mostly an observation (sorry if unclear, but I tried to indicate in review summary that this was not a blocking comment).

Reply:
If we ever get to doing a "happy eyeballs" implementation, we likely will want to re-use this, so we can re-use the `net.Dialer`'s implementation, and just have a mocked resolver that provides exactly the addresses we want...

Reply:
Now that we have made policy handle the `network` setting as well, I think testing end-to-end makes even more sense, so I went ahead and did that.